### ISSUE TITLE
[a11y-defect-4]: Date of birth field allows for a future date

### ISSUE THREAD

**Issue by tbaker1026** [department-of-veterans-affairs/va.gov-team#63439]
### Point of contact

Jeana Clark

### Severity level

4, Minor. Consider fixing or exploring in 2-4 sprints post-launch.

### Details

...
In the date of birth field you can add a future date, this should be prevented.  Found in 508 audit on form 21-4142

https://staging.va.gov/supporting-forms-for-claims/release-information-to-va-form-21-4142

### Reproduction steps

There is no validation when you enter a future date in the Date of birth field . For example, On the “Step 1 of 12” screen, you can enter “2123” in the Date of birth “Year” textbox and user can proceed to the next screen.
...
<img width="675" alt="Five" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/115116356/40bb6405-4bcd-46ad-9560-06ae768571fb">

### Proposed solution or next steps

...

### References, articles, or WCAG support

1.
2.
3.
...

### Type of issue

- [ ] Screenreader
- [ ] Keyboard
- [ ] Focus
- [ ] Headings
- [ ] Color
- [ ] Zoom
- [ ] Semantics
- [ ] Axe-core
- [ ] Something else
